lib/parsers/moov_parser/decoder.rb

```diff
@@ -77,16 +77,63 @@ def find_video_trak_atom(atoms)
     end
   end
 
+  # Recursive descent parser - will drill down to atoms which
+  # we know are permitted to have leaf/branch atoms within itself,
+  # and will attempt to recover the data fields for leaf atoms
+  #
+  # @param [IO] io
+  # @param [Integer] max_read
+  # @param [Array<String>] current_branch
+  # @return [Array<Atom>]
+  def extract_atom_stream(io, max_read, current_branch = [])
+    initial_pos = io.pos
+    atoms = []
```

Comment:
You could also use `each_with_object` to shorten declaration+block+return to just block, if it makes more sense to you 😊

Reply:
This is untouched from its previous implementation, I think it's just been relocated within the file as a side effect of old changes that I replaced with this dedicated ISO decoder approach. Idea is that this will be replaced down the line anyway, so I don't see a need to mess with it - good tip though!

Reply:
In fact, I'll revert all changes to the MOOV decoder, to be extra-sure.